azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/domain/VirtualMachineScaleSetOSProfile.java

```diff
@@ -0,0 +1,280 @@
+   public abstract static class WindowsConfiguration {
+
+      @AutoValue
+      public abstract static class WinRM {
+         public enum Protocol {
+
+            HTTP("http"),
+            HTTPS("https"),
+            UNRECOGNIZED("Unrecognized");
+
+            private String value;
+
+            Protocol(String value) {
+               this.value = value;
+            }
+
+            public static Protocol fromValue(String value) {
+               return (Protocol) GetEnumValue.fromValueOrDefault(value, Protocol.UNRECOGNIZED);
+            }
+
+            @Override
+            public String toString() {
+               return this.value;
+            }
+         }
+
+         @AutoValue
+         public abstract static class ProtocolListener {
+
+            /**
+             * The protocol for the protcol listener
+             */
+            public abstract Protocol protocol();
+
+            /**
+             * The certificate url or the protcol listener
+             */
+            @Nullable
+            public abstract String certificateUrl();
+
+            @SerializedNames({"protocol", "certificateUrl"})
+            public static ProtocolListener create(final Protocol protocol, final String certificateUrl) {
+
+               return new AutoValue_VirtualMachineScaleSetOSProfile_WindowsConfiguration_WinRM_ProtocolListener(
+                  protocol, certificateUrl);
+            }
+         }
+
+         /**
+          * Map of different settings
+          */
+         public abstract List<ProtocolListener> listeners();
+
+         @SerializedNames({"listeners"})
+         public static WinRM create(final List<ProtocolListener> listeners) {
+            return new AutoValue_VirtualMachineScaleSetOSProfile_WindowsConfiguration_WinRM(listeners == null ? ImmutableList.<ProtocolListener>of() : ImmutableList.copyOf(listeners));
+         }
+      }
+
+      @AutoValue
+      public abstract static class AdditionalUnattendContent {
+
+         public abstract String pass();
+
+         public abstract String component();
+
+         public abstract String settingName();
+
+         public abstract String content();
+
+         @SerializedNames({"pass", "component", "settingName", "content"})
+         public static AdditionalUnattendContent create(final String pass, final String component,
+                                                        final String settingName,
+                                                        final String content) {
+
+            return new AutoValue_VirtualMachineScaleSetOSProfile_WindowsConfiguration_AdditionalUnattendContent(
+               pass, component, settingName, content);
+         }
+      }
+
+      /**
+       * The provision VM Agent true of false.
+       */
+      public abstract boolean provisionVMAgent();
+
+      /**
+       * winRM
+       */
+      @Nullable
+      public abstract WinRM winRM();
+
+      /**
+       * unattend content
+       */
+      @Nullable
```

Comment:
You are enforcing presence and immutability. Remove this annotation.

Reply:
These @Nullable annotations are required to serialized the json response from the service (especially the when the value is not returned as sometimes there are not setup by the users)